### PR TITLE
Keeper v0.6.12 and some Aquarius related improvements.

### PR DESCRIPTION
**Pull request by Pedro-vk**
## Description

Upgraded Keeper contracts to v0.6.12.
Added `checksum` (inside `Metadata` service) generation on DDO.
Added `proof` generation on DDO.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [x] Follows the code style of this project.
- [x] Tests Cover Changes
- [x] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: src/ocean/OceanAssets.ts [hunk 1]
```diff
@@ -108,45 +103,59 @@ export default class OceanAssets {
...
+
         const storedDdo = await aquarius.storeDDO(ddo)
 
         // Logger.log(JSON.stringify(storedDdo, null, 2))
 
         await didRegistry.registerAttribute(
             did.getId(),
-            ValueType.URL,
-            "Metadata",
+            ddo.getChecksum().substr(0, 32), // TODO: checksum
```
Comment (Pedro-vk) [inline src/ocean/OceanAssets.ts]: @aaitor following the checksum generation specification is correct to take the first 32 bytes?
https://github.com/oceanprotocol/OEPs/tree/master/7#how-to-compute-the-integrity-checksum

File: library.json [hunk 2]
```diff
@@ -7,11 +7,11 @@
   "dependencies": [
     {
       "name": "keeper-contracts",
-      "version": "~0.5.3"
+      "version": "~0.6.11"
```
Comment (ssallam) [inline library.json]: Should this be `0.6.12`?

File: src/ocean/OceanAssets.ts [hunk 3]
```diff
@@ -72,9 +66,10 @@ export default class OceanAssets {
         const computeServiceDefintionId: string = "1"
         const metadataServiceDefinitionId: string = "2"
 
-        metadata.base.contentUrls =
-            [await SecretStoreProvider.getSecretStore()
-                .encryptDocument(did.getId(), metadata.base.contentUrls)]
+        metadata.base.encryptedFiles = [
+            await SecretStoreProvider.getSecretStore()
+                .encryptDocument(did.getId(), metadata.base.files),
+        ]
```
Comment (ssallam) [inline src/ocean/OceanAssets.ts]: In squid-py this is not a list.

Comment (Pedro-vk) [reply src/ocean/OceanAssets.ts]: There is no information in the OEP, so I'm going to move to your implementation.

File: test/utils/SignatureHelpers.test.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        it("should sign a text as expected using password", async () => {
+            const signed = await signText(text, publicKey, "test")
+
+            assert.equal(signed, signature)
+            expect(personalSignSpy).to.have.been.called.with(text, publicKey, "test")
+        })
+    })
+
+    describe("#verifyText", () => {
+        it("should recover the privateKey of a signed message", async () => {
```
Comment (ssallam) [inline test/utils/SignatureHelpers.test.ts]: Rather recovers `publicKey`

**Pull request merged**